internal/scripts/crates_io_publish_script.sh

```diff
@@ -32,3 +32,5 @@ cargo publish -p iceoryx2-bb-memory
 cargo publish -p iceoryx2-cal
 cargo publish -p iceoryx2-bb-trait-tests
 cargo publish -p iceoryx2
+cargo publish -p iox2
```

Comment:
If we publish this and the services, we also need a little readme describing how to install them. I think `cargo install iox2` and `cargo install iox2-services` should work. This could be placed into `iceoryx2-cli/README.md`

Reply:
@elfenpiff It's not clear exactly what you expect. Do you have an example?

Reply:
I try to phrase it as a use case so that it is better understandable.

* **As a** iceoryx2 user, 
* **that** uses iceoryx2 as a dependency in my projects workspace `Cargo.toml`
* **I want** install the `iox2` cli
* **So that** I can list all services and nodes.

So the question is, how does the iceoryx2 user install `iox2` cli with all of its subcommand.
1. `cargo run --bin iox2 -- services list` does not work -> `Failed to execute command: Command not found: services`
2. `cargo run --bin iox2-services -- list` -> works but is not the intended use case.

So I would like to have a `README.md` document inside `iceoryx2-cli/iox2`, since this is the document that is shown in crates.io when the crate is published, see: https://crates.io/crates/iceoryx2

In this document there is a installation guide with terminal commands on how to install the `iox2` cli with all of its subcommands so that the subcommands are installed in the right path and the user can just use it as intended.

I assume that the intended use case is, that `iox2` is in the `PATH` of the shell and the user can just call
```sh
iox2 services list
```
for instance.

Reply:
@elBoberido whats your take on this?

Reply:
I guess you mean something like https://crates.io/crates/cargo-outdated or https://crates.io/crates/cargo-list. Without a README.md in the folder, the page on crates.io will look like this https://crates.io/crates/iceoryx2-bb-container

I think havin a small README.md for the cli tools would be helpful for people looking for them on crates.io. It does not need to be fancy in the beginning, just a hint on how to install and use them, maybe with an output of a cli tool as example.

Reply:
@elfenpiff @elBoberido I think I understand my confusion now, there are actually two things being asked:

1. How does a user install the `iox2` CLI tools
2. How does the user use the CLI tools

For (1) I can add a short readme to the crates, but it will essentially just boil down to `cargo install iox2`. I did some research and might need to tweak the `Cargo.tml` file slightly to make this work.

For (2), the user should use `iox2 --help`. Writing documentation (like in https://crates.io/crates/cargo-list) will be tedious to maintain IMO. I can add the instruction for user to use `iox2 --help` (or equivalent) to the readmes as a compromise.

Reply:
> So the question is, how does the iceoryx2 user install iox2 cli with all of its subcommand.
> 
> cargo run --bin iox2 -- services list does not work -> Failed to execute command: Command not found: services
> cargo run --bin iox2-services -- list -> works but is not the intended use case.

I replied with the reasoning for this in a comment in another PR, but if you do `cargo run --bin iox2`, the reason it did not work will be clear.

Reply:
@orecham I just want to have a readme for `1. How does the user install the iox2 CLI tools`. For `2. How does the user use the CLI tools` the user can call `iox2 --help`.

Reply:
I think a short description at the top what the tool does would be helpful, together with a link to the iceoryx2 project. Followed by how to install and how to use. For the how to use section, the most common usage should be shown, e.g. `iox2 services` and how to discover more options, e.g. `iox servives --help`. The output of the most common command would also be nice but not a must have.

Reply:
@elfenpiff @elBoberido I moved all tools into a single crate for easy installation and added a short readme for that crate

you can test installation locally:

```
$ cargo install --path ./iceoryx2-cli
```

users should simply be able to do:

```
$ cargo install iceoryx2_cli
```